crates/common/src/utils.rs

```diff
@@ -54,3 +56,20 @@ pub fn ignore_metadata_hash(bytecode: &[u8]) -> &[u8] {
         bytecode
     }
 }
+
+/// Strips all __$xxx$__ placeholders from the bytecode if it's an unlinked bytecode.
+/// by replacing them with 20 zero bytes.
+/// This is useful for matching bytecodes to a contract source, and for the source map,
+/// in which the actual address of the placeholder isn't important.
+pub fn strip_bytecode_placeholders(bytecode: &BytecodeObject) -> Option<Bytes> {
+    match &bytecode {
+        BytecodeObject::Bytecode(bytes) => Some(bytes.clone()),
+        BytecodeObject::Unlinked(s) => {
+            // Replace all __$xxx$__ placeholders with 32 zero bytes
+            let re = Regex::new(r"__\$.{34}\$__").expect("invalid regex");
```

Comment:
imo we can keep this for this pr as is but when we move this, we should make this a lazy static